src/webui/api/torrentscontroller.cpp

```diff
@@ -559,12 +559,20 @@ void TorrentsController::filePrioAction()
     checkParams({"hash", "id", "priority"});
 
     const QString hash = params()["hash"];
-    int fileID = params()["id"].toInt();
-    int priority = params()["priority"].toInt();
-    BitTorrent::TorrentHandle *const torrent = BitTorrent::Session::instance()->findTorrent(hash);
+    const QStringList fileIDs {params()["id"].split('|')};
+    bool ok;
+    const int priority = params()["priority"].toInt(&ok);
```

Comment:
The `FilePriority` enum is defined in `gui/torrentcontentmodelitem.h`, which we can't import into a web controller. Any thoughts on where we should migrate this enum? I would think somewhere inside `base` but I don't know those classes very well.

Reply:
I think you missed my point. The valid priority values are defined in `gui/torrentcontentmodelitem.h`, which is the list I can check against to see if the value will be accepted.

Reply:
> offtopic, it seems torrent->setFilePriority only changes 1 prio at a time which is inefficient when you invoke it in a loop.
 
My latest commit modifies `TorrentHandle::setFilePriority()`to take a list of indexes, since the controller code is the only place we call it.

Reply:
>The valid priority values are defined in gui/torrentcontentmodelitem.h, which is the list I can check against to see if the value will be accepted.

That is the gui part, but really who is consuming the priority values? libtorrent, you should check from there.

Reply:
It's not just for the gui, because we use the same priority values in the webui. The values are defined in gui code because we have very little separation between gui logic and application code, but they really should be moved to base.

libtorrent will accept a priority between 0-7 ([source](https://www.libtorrent.org/reference-Core.html#piece-priority-prioritize-pieces-piece-priorities)); we currently only use 0, 1, 6, and 7. Without the values in `gui/torrentcontentmodelitem.h`, users would be able to enter a value between 2-5.

I'm moving the `prio` namespace to a new file `base/torrentpriority.h`. If we should move it, lmk.

Reply:
>That is the gui part, but really who is consuming the priority values? libtorrent, you should check from there.

You shouldn't use libtorrent interface directly outside of abstraction layer (BitTorrent:: types and functions).
>If we should move it, lmk.

Yes. Torrent file priority enum should be declared in BitTorrent namespace. Maybe in torrenthandle.h file (I'm not sure there is separate file should exist).

Reply:
>My latest commit modifies `TorrentHandle::setFilePriority()` to take a list of indexes, since the controller code is the only place we call it.

Isn't `prioritizeFiles()` enough?

Reply:
We still have to do some extra accounting since `prioritizeFiles()` wants a list of priorities for all files in the torrent. But I think that's very doable and will allow us to get rid of `TorrentHandle::setFilePriority()` entirely.